hs_core/views/oauth2_view.py

```diff
@@ -0,0 +1,22 @@
+from oauth2_provider.views.base import AuthorizationView
+from django.contrib.auth.mixins import AccessMixin
+from django.contrib.auth.models import User
+from django.shortcuts import redirect
+
+
+class GroupRequiredMixin(AccessMixin):
+    """
+    Verify that the current user is in the specified group.
+    Redirects to authorization group page if user is not part of the group.
+    """
+    def dispatch(self, request, *args, **kwargs):
+        u = User.objects.get(pk=self.request.user.id)
+
+        if 1 != len(u.uaccess.my_groups.filter(id=kwargs['group_id'])):
```

Comment:
@sblack-usu Would ```exists()``` on the filter queryset serve the same purpose as checking its length? If so, using ```exists()``` would be better and more readable.

Reply:
good idea, I can do that.